recipes/cspice/all/conanfile.py

```diff
@@ -0,0 +1,160 @@
+import os
+import time
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class CspiceConan(ConanFile):
+    name = "cspice"
+    description = "NASA C SPICE library"
+    license = "TSPA"
+    topics = ("conan", "spice", "naif", "kernels", "space", "nasa", "jpl", "spacecraft", "planet", "robotics")
+    homepage = "https://naif.jpl.nasa.gov/naif/toolkit.html"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "patches/**"]
+    exports = ["TSPA.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    @property
+    def _patch_per_platform(self):
+        return {
+            "Macos": {
```

Comment:
the recipe uses lots of patches.
it would be nice if you could add some comment(s) about them, maybe links where did you take em (issue, commit, mail list, etc.), and some descriptions why are they needed / what do they do

Reply:
CSpice is generated from fortran by f2c. There are few differences in source code depending on triplet. Since the recipe can't conditionnaly download the source code depending on triplet, it downloads the Cygwin 32 bits package, and after that it applies these patches at build time to handle these differences (I have created these patches myself, there is no correction of issues etc).
I can add a comment about all of this.